tests/integration/Storefront/Checkout/Cart/SalesChannel/StorefrontCartFacadeTest.php

```diff
@@ -0,0 +1,184 @@
+use Shopware\Core\Checkout\Payment\PaymentMethodEntity;
+use Shopware\Core\Checkout\Shipping\Cart\Error\ShippingMethodBlockedError;
+use Shopware\Core\Checkout\Shipping\ShippingMethodEntity;
+use Shopware\Core\Content\Test\Product\ProductBuilder;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Test\TestCaseBase\IntegrationTestBehaviour;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Core\System\SalesChannel\Context\SalesChannelContextPersister;
+use Shopware\Core\System\SalesChannel\Context\SalesChannelContextService;
+use Shopware\Core\System\SalesChannel\Context\SalesChannelContextServiceParameters;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Shopware\Core\Test\Stub\Framework\IdsCollection;
+use Shopware\Core\Test\TestDefaults;
+use Shopware\Storefront\Checkout\Cart\Error\PaymentMethodChangedError;
+use Shopware\Storefront\Checkout\Cart\Error\ShippingMethodChangedError;
+use Shopware\Storefront\Checkout\Cart\SalesChannel\StorefrontCartFacade;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+class StorefrontCartFacadeTest extends TestCase
+{
+    use IntegrationTestBehaviour;
+
+    private IdsCollection $ids;
+
+    private SalesChannelContext $context;
+
+    private CartService $cartService;
+
+    private StorefrontCartFacade $cartFacade;
+
+    protected function setUp(): void
+    {
+        $this->ids = new IdsCollection();
+        $this->cartService = $this->getContainer()->get(CartService::class);
+        $this->cartFacade = $this->getContainer()->get(StorefrontCartFacade::class);
+    }
+
+    public function testBlockedShippingMethodMessage(): void
+    {
+        $id = $this->changeShippingMethodAvailabilityRuleId();
+
+        static::getContainer()->get(SalesChannelContextPersister::class)
+            ->save($this->ids->get('token'), ['shippingMethodId' => $id], TestDefaults::SALES_CHANNEL);
+
+        $this->context = self::getContainer()
+            ->get(SalesChannelContextService::class)
+            ->get(new SalesChannelContextServiceParameters(TestDefaults::SALES_CHANNEL, $this->ids->get('token')));
+
+        $cart = new Cart($this->ids->get('token'));
+        $cart = $this->cartService->add($cart, $this->createProduct(), $this->context);
+
+        static::assertInstanceOf(ShippingMethodBlockedError::class, $cart->getErrors()->first());
+
+        $result = $this->cartFacade->get($this->ids->get('token'), $this->context);
+
+        static::assertCount(1, $result->getErrors());
+
+        $error = $result->getErrors()->first();
+
+        static::assertInstanceOf(ShippingMethodChangedError::class, $error);
+        static::assertSame('Standard', $error->getOldShippingMethodName());
+        static::assertSame('Express', $error->getNewShippingMethodName());
```

Comment:
Could you please, at the end of both tests, fetch a new SalesChannelContext from the Service and also assert, that the other payment method is actually selected?